src/algorithms/synthesis/syrec_synthesis.cpp

```diff
@@ -990,7 +990,7 @@ namespace syrec {
 
             // types of constant and garbage
             constant constVar = (var->type == Variable::Out || var->type == Variable::Wire) ? constant(false) : constant();
-            bool     garbage  = (var->type == Variable::In || var->type == Variable::Wire);
+            bool     garbage  = var->type == Variable::Wire;
```

Comment:
Why is an input-only variable no longer considered garbage?

Reply:
I assumed we considered a signal to be garbage if the value that is store by the signal is not required/useful right? Since in contains value which is required, I thought I would remove it.

Reply:
If a variable is _just_ an input variable then it is per definition a garbage output since you don't care for the result.

Reply:
Oh okay, I will add the in signal back then.